app/models/nta_upload.rb

```diff
@@ -273,6 +300,244 @@ def NtaUpload.load(id, series_id)
+      insert measurement_series (measurement_id, series_id)
+      select distinct m.id, s.id
+      from series s
+        join geographies g on g.id = s.geography_id
+        join measurements m
+           on m.universe = s.universe
+          and m.prefix = concat(substring_index(s.name, '@', 1), '_incgrp2015')
+      where s.universe = 'NTA'
+      and g.geotype = 'incgrp1'
+    SQL
+    puts "DEBUG: load_cats_postproc ASSOCIATE data list meas indent0 at #{Time.now}"
+    NtaUpload.connection.execute <<~SQL
+      /*** Associate all measurements NTA_<var>_{regn,incgrp2015} with their data lists at indent 0 ***/
+      insert data_list_measurements (data_list_id, measurement_id, indent)
+      select distinct dl.id, m.id, 'indent0'
+      from data_lists dl
+        join measurements m
+           on dl.name = substring_index(m.prefix, '_', 2)
+          and length(m.prefix)-length(replace(m.prefix, '_', '')) = 2
+          and dl.universe = m.universe
+      where dl.universe = 'NTA'
+    SQL
+    puts "DEBUG: load_cats_postproc ASSOCIATE data list meas indent1 at #{Time.now}"
+    NtaUpload.connection.execute <<~SQL
+      /*** Associate all measurements NTA_<var>_{regn,incgrp2015}_{subcategory} with their data lists at indent 1 ***/
+      insert data_list_measurements (data_list_id, measurement_id, indent)
+      select distinct dl.id, m.id, 'indent1'
+      from data_lists dl
+        join measurements m
+           on dl.name = substring_index(m.prefix, '_', 2)
+          and length(m.prefix)-length(replace(m.prefix, '_', '')) > 2
+          and dl.universe = m.universe
+      where dl.universe = 'NTA'
+    SQL
+    puts "DEBUG: load_cats_postproc AGGREGATE data points at #{Time.now}"
+    NtaUpload.connection.execute <<~SQL
+      /*** Generating aggregate region/income group data points ***/
+      insert data_points (universe, series_id, data_source_id, created_at, `date`, `value`)
+      select distinct dp.universe, s2.id, ds.id, now(), dp.`date`, avg(dp.`value`)
+      from data_points dp
+        join series s1 on dp.series_id = s1.id  /* country data series */
+        join geographies g on g.id = s1.geography_id
+        join series s2 on s2.universe = s1.universe  /* aggregate region/incgrp series */
+        join geo_trees gt
+           on gt.child_id = s1.geography_id
+          and gt.parent_id = s2.geography_id
+        join data_sources ds on ds.series_id = s2.id
+
+        join measurement_series ms1 on s1.id = ms1.series_id
+        join measurements        m1 on m1.id = ms1.measurement_id
+        join data_list_measurements dm1 on dm1.measurement_id = m1.id
+
+        join measurement_series ms2 on s2.id = ms2.series_id
+        join measurements        m2 on m2.id = ms2.measurement_id
+        join data_list_measurements dm2 on dm2.measurement_id = m2.id
+      where dp.universe = 'NTA'
+      and g.geotype = 'region3'
+      and dm1.data_list_id = dm2.data_list_id
+      group by 1,2,3,4,5
```

Comment:
You can enclose the remaining non-aggregated columns in `ANY_VALUE()` (https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_only_full_group_by, https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html). Then you can use the easier to read `group by` statement.